File: Swift/twosum.swift
func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var numToIndex = [Int: Int]()
    
    for (index, num) in nums.enumerated() {
        let complement = target - num
        if let complementIndex = numToIndex[complement] {
            return [complementIndex, index]
        }
        numToIndex[num] = index
    }
    
    return []
}

File: Rust/twosum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_to_index = HashMap::new();
        
        for (index, num) in nums.iter().enumerate() {
            let complement = target - num;
            if let Some(&complement_index) = num_to_index.get(&complement) {
                return vec![complement_index as i32, index as i32];
            }
            num_to_index.insert(num, index as i32);
        }
        
        Vec::new()
    }
}
